### PR TITLE
chore: add a function to flattern tags for AUTO

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud' TESTARGS='-run=TestAccSomethingV0_basic'
...
=== RUN   TestAccSomethingV0_basic
--- PASS: TestAccSomethingV0_basic (70.75s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud       70.796s
```

### CODE REVIEW

File: huaweicloud/utils/tags.go [hunk 1]
```diff
@@ -68,6 +68,21 @@ func TagsToMap(tags []tags.ResourceTag) map[string]string {
 	return result
 }
 
+// FlattenTagsToMap returns the list of tags into a map.
+func FlattenTagsToMap(tags interface{}) map[string]string {
```
Comment (ShiChangkuo) [inline huaweicloud/utils/tags.go:72]: can the type be **[]interface{}**, then line 76 can removed

File: huaweicloud/utils/tags.go [hunk 2]
```diff
@@ -68,6 +68,21 @@ func TagsToMap(tags []tags.ResourceTag) map[string]string {
...
+	if tags == nil {
+		return nil
+	}
+	tagArray := tags.([]interface{})
+	result := make(map[string]string)
+	for _, val := range tagArray {
+		t := val.(map[string]string)
```
Comment (ShiChangkuo) [inline huaweicloud/utils/tags.go]: using `t, ok := val.(map[string]string)` will be better

File: huaweicloud/utils/tags.go (hunk 1 above)
Comment (hmbig2) [reply huaweicloud/utils/tags.go:72]: type interface is more convenient.

File: huaweicloud/utils/tags.go (hunk 2 above)
Comment (hmbig2) [reply huaweicloud/utils/tags.go]: OK, fixed

**Pull request merged**